lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -1858,7 +1879,9 @@ class MetricsEndpointAggregator(Object):
     constructing an aggregator object.
     """
 
-    def __init__(self, charm, relation_names, relabel_instance=True):
+    _stored = StoredState()
+
+    def __init__(self, charm, relation_names: Optional[dict] = None, relabel_instance=True):
```

Comment:
Wdyt about typing `relation_names` as TypedDict?
And document that `downstream-prometheus-scrape`,  etc. are the default?

Reply:
`downstream-prometheus-scrape` as the default is already [clearly documented](https://github.com/canonical/cos-proxy-operator) in what will be the only consumer ever.